mat64/symmetric.go

```diff
@@ -5,16 +5,41 @@ import (
 	"github.com/gonum/blas/blas64"
 )
 
-// Symmetric represents a symmetric matrix.
-type Symmetric struct {
+var (
+	symDense *SymDense
+
+	_ Matrix          = symDense
+	_ Symmetric       = symDense
+	_ RawSymmmetricer = symDense
+)
+
+const (
+	ErrUplo = "mat64: blas64.Symmetric not upper"
+)
+
+// SymDense is a symmetric matrix that uses Dense storage.
+type SymDense struct {
 	mat blas64.Symmetric
 }
 
-// NewSymmetric constructs an n x n symmetric matrix. If len(mat) == n * n,
+// Symmetric represents a symmetric matrix (where the element at {i, j} equals
+// the element at {j, i}). Symmetric matrices are always square.
+type Symmetric interface {
+	Matrix
+	// Symmetric returns the number of rows/columns in the matrix.
+	Symmetric() int
```

Comment:
Agreed :)

> On Feb 2, 2015, at 3:29 PM, Dan Kortschak <notifications@github.com> wrote:
> 
> In mat64/symmetric.go <https://github.com/gonum/matrix/pull/95#discussion_r23969357>:
> 
> >  	mat blas64.Symmetric
> >  }
> >  
> > -// NewSymmetric constructs an n x n symmetric matrix. If len(mat) == n * n,
> > +// Symmetric represents a symmetric matrix (where the element at {i, j} equals
> > +// the element at {j, i}). Symmetric matrices are always square.
> > +type Symmetric interface {
> > +	Matrix
> > +	// Symmetric returns the number of rows/columns in the matrix.
> > +	Symmetric() int
> Or then again... maybe not. That would fail to distinguish Symmetric from Triangular.
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/gonum/matrix/pull/95/files#r23969357>.
>